src/invidious/helpers/youtube_api.cr

```diff
@@ -67,6 +67,48 @@ def request_youtube_api_browse(browse_id : String, params : String, region : Str
   return _youtube_api_post_json("/youtubei/v1/browse", data)
 end
 
+####################################################################
+# request_youtube_api_next(continuation)
+# request_youtube_api_next(video_id, params)
+#
+# Requests the youtubei/v1/next endpoint with the required headers
+# and POST data in order to get a JSON reply in english that can
+# be easily parsed.
+#
+# The requested data can either be:
+#
+#  - A continuation token (ctoken). Depending on this token's
+#    contents, the returned data can be comments, playlist videos,
+#    search results, channel community tab, ...
+#
+#  - A video ID (parameters MUST be an empty string)
+#
+
+def request_youtube_api_next(continuation : String)
+  # JSON Request data, required by the API
+  data = {
+    "context"      => make_youtube_api_context("US"),
+    "continuation" => continuation,
+  }
+
+  return _youtube_api_post_json("/youtubei/v1/next", data)
+end
+
+def request_youtube_api_next(video_id : String, params : String)
+  # JSON Request data, required by the API
+  data = {
+    "videoId" => video_id,
+    "context" => make_youtube_api_context("US"),
+  }
+
+  # Append the additionnal parameters if those were provided
+  if params != ""
+    data["params"] = params
+  end
```

Comment:
can the next endpoint take parameters?

Reply:
I have seen some requests using it while browsing the YouTube app for Android, but not on the desktop interface(www.youtube.com). I'm leaving this part of code for the moment because we may have to use it later (with #2189) if that's okay for you.

Reply:
Okay, so we keep it in here :)